cosmwasm/packages/sgx-vm/src/enclave_tests.rs

```diff
@@ -0,0 +1,10 @@
+// I reexport `sgx_types` here so that `go-cosmwasm` doesn't need to specify it as a dev-dependency.
+pub use sgx_types;
+use sgx_types::{sgx_enclave_id_t, sgx_status_t};
+
+pub use crate::enclave::get_enclave;
+
+#[link(name = "rust_cosmwasm_enclave.signed")]
+extern "C" {
+    pub fn ecall_run_tests(eid: sgx_enclave_id_t) -> sgx_status_t;
```

Comment:
To be able to run this in CI we would need this to return an exit code. Could you make it so the function returns an integer or something so we have a success/fail indicator?

Reply:
I'd ideally like to return a full test report. So, per test whether it succeeded or failed (panicked).
That would require an ocall or something though, since it'll have a dynamic size and i don't like that...
How about returning the number of failed tests, so if no tests fail, we'll get the nice "0" which is also a success status?

Reply:
Yep, returning the amount of failures should be fine. That's usually what test runners do afaik